File: SDS.Clould/SDS.Clould.WebAPI/DataModels/NewPackageDataModel.cs
namespace SDS.Clould.WebAPI.DataModels
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel.DataAnnotations;
    using System.Linq;

    using SDS.Clould.Models;

    public class NewPackageDataModel
    {
                                                                                   // taken from the model or passed
        public static Package FromModel(NewPackageDataModel model, string ownerId, string pictureUrl)
        {
            return new Package()
            {
                Deadline = model.Deadline,
                FromTown = model.FromTown,
                Kilograms = model.Kilograms,
                Notes = model.Notes,
                Price = model.Price,
                Space = model.Space,
                ToTown = model.ToTown,
                UserId = ownerId,
                PictureURL = pictureUrl
            };
        }

        [Required]
        public string Picture { get; set; }

        public decimal Price { get; set; }

        // in square meters
        public int Space { get; set; }

        public int Kilograms { get; set; }

        [Required]
        public string FromTown { get; set; }

        [Required]
        public string ToTown { get; set; }

        public DateTime Deadline { get; set; }

        public string Notes { get; set; }
    }
}

File: SDS.Mobile/app/src/main/java/com/example/spawneddeliveryservice/models/NewPackageDataModel.java
package com.example.spawneddeliveryservice.models;

public class NewPackageDataModel {
    public String Picture;
    public double Price;
    public int Space;
    public int Kilograms;
    public String FromTown;
    public String ToTown;
    public String Deadline;//2015-03-17 format
    public String Notes;

    public NewPackageDataModel(String base64Picture, double price, int space, int kilograms, String fromTown, String toTown, String deadline, String notes) {
        this.Picture  = base64Picture;
        this.Price = price;
        this.Space = space;
        this.Kilograms = kilograms;
        this.FromTown = fromTown;
        this.ToTown = toTown;
        this.Deadline = deadline;
        this.Notes = notes;
    }
}
